### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by ipavlopoulos**
Refactored to join the SemEval21 and ACL22 work.

### CODE REVIEW

File: SemEval2021/baselines/models.py [hunk 1]
```diff
@@ -277,4 +283,94 @@ def tune_threshold(self, validation_data, evaluator):
...
+
+    def fit(self, X_train, y_train, X_dev=None, y_dev=None):
+        train_vectors = self.vectorizer.fit_transform(X_train)
+        # self.tfidf_vectorizer.fit(train_vectors)
+        # train_vectors = self.tfidf_vectorizer.transform(train_vectors)
+        self.model.fit(train_vectors, y_train)
+
+    def predict(self, x):
+        test_vectors = self.vectorizer.transform(x)
+        # test_vectors = self.tfidf_vectorizer.transform(counts)
```
Comment (sorensenjs) [inline SemEval2021/baselines/models.py]: remove this comment

File: SemEval2021/baselines/models.py [hunk 2]
```diff
@@ -277,4 +283,94 @@ def tune_threshold(self, validation_data, evaluator):
...
+                 lowercase=True):
+        self.tokenise = tokenise
+        self.use_idf = use_idf
+        self.vectorizer = CountVectorizer(analyzer='word',
+                                          stop_words="english",
+                                          tokenizer=tokenise,
+                                          lowercase=lowercase,
+                                          binary=binary,
+                                          max_features=max_features)
+        # self.tfidf_vectorizer = TfidfTransformer(use_idf=self.use_idf)
```
Comment (sorensenjs) [inline SemEval2021/baselines/models.py]: remove

File: SemEval2021/baselines/models.py [hunk 3]
```diff
@@ -277,4 +283,94 @@ def tune_threshold(self, validation_data, evaluator):
...
+        self.model = model
+
+    def fit(self, X_train, y_train, X_dev=None, y_dev=None):
+        train_vectors = self.vectorizer.fit_transform(X_train)
+        # self.tfidf_vectorizer.fit(train_vectors)
+        # train_vectors = self.tfidf_vectorizer.transform(train_vectors)
```
Comment (sorensenjs) [inline SemEval2021/baselines/models.py]: remove these

File: SemEval2021/baselines/models.py [hunk 4]
```diff
@@ -6,13 +6,19 @@
...
+from tensorflow.keras.preprocessing.sequence import pad_sequences
+from tensorflow.keras.models import Model
+from tensorflow.keras.layers import Input
+from tensorflow.keras.layers import GRU, LSTM, Embedding, Dense, TimeDistributed, Dropout, Bidirectional
+from tensorflow.keras.callbacks import EarlyStopping
+from tensorflow.keras.utils import plot_model
+from tensorflow.keras.metrics import BinaryAccuracy, Precision, Recall, AUC
+from sklearn.model_selection import train_test_split
+from sklearn.utils import shuffle
+from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer
```
Comment (sorensenjs) [inline SemEval2021/baselines/models.py]: remove TfidfTransformer

**Review by sorensenjs — APPROVED** [review]

**Review by LeoLaugier — APPROVED** [review]

File: SemEval2021/baselines/models.py (hunk 1 above)
Comment (ipavlopoulos) [reply SemEval2021/baselines/models.py]: Removed.

File: SemEval2021/baselines/models.py (hunk 2 above)
Comment (ipavlopoulos) [reply SemEval2021/baselines/models.py]: Removed.

File: SemEval2021/baselines/models.py (hunk 3 above)
Comment (ipavlopoulos) [reply SemEval2021/baselines/models.py]: Removed.

File: SemEval2021/baselines/models.py (hunk 4 above)
Comment (ipavlopoulos) [reply SemEval2021/baselines/models.py]: Removed.

**Pull request merged**